api/test/wfl/integration/datarepo_test.clj

```diff
@@ -59,3 +61,28 @@
                  (datarepo/ingest-table dataset-id table-url table-name))]
             (is (= 1 row_count))
             (is (= 0 bad_row_count))))))))
+
+(deftest test-create-snapshot
+  ;; To test given a dataset, we can query BigQuery for row-ids
+  ;; and use them to create a snapshot!
+  (let [tdr-profile (env/getenv "WFL_TDR_DEFAULT_PROFILE")
+        definition "sarscov2-illumina-full-inputs.json"]
+    (testing "creating snapshots from dataset"
+      (fixtures/with-temporary-dataset
+        (datasets/unique-dataset-request tdr-profile definition)
+        (fn [dataset-id]
+          (let [{:keys [dataProject] :as dataset} (datarepo/dataset dataset-id)
+                table     "sarscov2_illumina_full_inputs"
+                today     (util/datetime->str (util/now))
+                yesterday (util/datetime->str (util/n-day-from-now -1))
```

Comment:
Any reason why you're not using https://docs.oracle.com/javase/8/docs/api/java/time/LocalDate.html? That will remove the need for the string formatting I think.

Reply:
If you want to use UTC, you can do this:
```clojure
(defn today-utc [] (java.time.LocalDate/now (java.time.Clock/systemUTC))
```

Reply:
I was thinking we will probably to have to support `TIMESTAMP`s plus `DATE`s if there are requirements on how many samples to be processed within a day. (and we don't know how many times / # of samples will be ingested everyday yet...) so I thought making it support timestamps from now might be a good idea.

Plus I didn't realize we use `OffsetDateTime/now` for out DB stuff right now looking at

 ![](https://i.stack.imgur.com/QPhGW.png)

I chose the UTC one… I can swap with either `LocalDate` or `OffsetDateTime`, what do you think?

Reply:
Based on offline discussion, will use `OffsetDateTime` for timestamps and `(str LocalDate)` for dates for date formatting. I will also remove the string formatting utilities as we don't use them.